### DISCUSSION TITLE
Help send custom emoji

### DISCUSSION THREAD

**Discussion by jekeam** [python-telegram-bot/python-telegram-bot#3375] [Q&A] [answered]
I'm trying to send custom emoji:
```
.send_message(
        chat_id=chat_id,
        text="😎hello from your telegram bot",
        entities=[
            MessageEntity(
                type=Sticker.CUSTOM_EMOJI,
                offset=0,
                length=2,
                custom_emoji_id="6046140249875156202",
            )]
    )
```

Why doesn't it work? the answer does not contain "entities", please tell me:
```

{
  "chat": {
    "id": ...,
    "type": <ChatType.PRIVATE>,
    "first_name": "...."
  },
  "text": "😎hello from your telegram bot",
  "group_chat_created": false,
  "entities": [],
  "new_chat_membeoup_chat_created": false,
  "photo": [],
  "channel_chat_created": false,
  "from": {rs": [],
  "new_chat_photo": [],
  "message_id": ....,
  "delete_chat_photo": false,
  "caption_entities": [],
  "date": ....,
  "supergroup_chat_created": false,
  "photo": [],
  "channel_chat_created": false,
  "from": {
    "is_bot": true,
    "username": "....",
    "first_name": ".",..
    "id": ..
  }
}
```

I will also be very grateful for an example of how to send premium animated emojis, for example: https://t.me/addemoji/WeatherEmoji

**Comment by Poolitzer** [comment, marked as answer]
You can not send them with a bot account as of now

**Reply by jekeam** [reply]
Can you tell me where to read about it? It seems they are in the documentation, why does it not work?

**Reply by clot27** [reply]
It is not supported by telegram bot api yet, keep an eye on t.me/botnews to get notified when it is added.

**Reply by Poolitzer** [reply]
> It seems they are in the documentation

For receiving them, not sending.

**Reply by BrokedTV** [reply]
I have seen those on a few bots

**Reply by Poolitzer** [reply]
yes premium bots can now

**Reply by BrokedTV** [reply]
can you share more informations about it? I would love to understand how to use custom emoji

**Reply by SsNiPeR1** [reply]
You can buy a username on Fragment, pay 5000 TON to make it assignable to a bot, and bot will be able to use Premium emoji.

**Reply by SsNiPeR1** [reply]
Any bot that is assigned with an NFT username is able to send Premium emoji. It doesn't matter who owns the username or the bot. And btw you don't purchase these usernames with your Telegram account, you purchase them with your TON wallet.

**Reply by masterpro-com** [reply]
But I've seen bots without additional usernames use the Custom Emoji. This is the case with several bots.

**Reply by clot27** [reply]
In just last update telegram added ability for all bots to send custom emojis as long as owner of those bots have subscribed to telegram premium, see: https://core.telegram.org/bots/api-changelog#february-9-2026

**Reply by hanhdth1211** [reply]
Quick follow-up: has anyone here with a Fragment-purchased username actually managed to send custom emoji into a channel? Owner Premium only covers private/group/supergroup, so I'm trying to confirm the Fragment route really works for channel posts before recommending it to a client. Thanks!

**Reply by SsNiPeR1** [reply]
> Quick follow-up: has anyone here with a Fragment-purchased username actually managed to send custom emoji into a channel? Owner Premium only covers private/group/supergroup, so I'm trying to confirm the Fragment route really works for channel posts before recommending it to a client. Thanks!

If the bot owner is subscribed to Telegram Premium, but the bot does not have an upgraded username assigned, it only can use Premium emoji in DMs, and they will be replaced with the fallback ordinary emoji upon sending the message to a channel.

**Reply by hanhdth1211** [reply]
Thanks! I'm actually considering buying a Fragment username for this, but before I spend on it I really want to confirm it works for channels specifically. Have you (or anyone) actually tested a Fragment username and seen custom emoji render in a channel post?
